plugins/pencilblue/controllers/actions/admin/sites/delete_site.js

```diff
@@ -53,8 +53,8 @@ module.exports = function DeleteSiteActionModule(pb) {
         var self = this;
         var siteid = self.pathVars.siteid;
         this.service.deleteSingle({where: {uid: siteid}}, function (err, site) {
-            if (util.isError(err)) {
-                return cb(err);
+            if (util.isError(err)|| !site) {
+                err = new Error("Failed to retrieve site from database for deletion");
```

Comment:
Per your email, you recommended a 404? I was working with this locally and prefer this flow. Displays a pop-up on site delete. Your call though. I have no problem doing a 404 instead.